packages/knip/src/ProjectPrincipal.ts

```diff
@@ -260,14 +261,30 @@ export class ProjectPrincipal {
   }
 
   public findUnusedMembers(filePath: string, members: ExportMember[]) {
-    if (!this.findReferences) {
+    if (!this.findReferences || !this.getImplementationAtPosition) {
       const languageService = ts.createLanguageService(this.backend.languageServiceHost, ts.createDocumentRegistry());
       this.findReferences = timerify(languageService.findReferences);
+      this.getImplementationAtPosition = timerify(languageService.getImplementationAtPosition);
     }
 
     return members.filter(member => {
       if (member.jsDocTags.has(PUBLIC_TAG)) return false;
-      const referencedSymbols = this.findReferences?.(filePath, member.pos) ?? [];
+      const implementations =
+        this.getImplementationAtPosition?.(filePath, member.pos)?.filter(
+          impl => impl.fileName !== filePath || impl.textSpan.start !== member.pos
```

Comment:
Being in the same file should not be a differentiator, right? Especially the abstract class and the class implementing it might be in the same file.